### DISCUSSION TITLE
how to join ldap user to vhost created in local ?

### DISCUSSION THREAD

**Discussion by spham** [rabbitmq/rabbitmq-server#3139] [Questions] [answered]
Hi
i need to know how to join user in ldap group to vhost created (dev, qa, or prod) 

do i need to create for each vhost a ldap group ?

```
ie. vhost dev => ldap ou = dev_rmq
```

so when I add user in dev_rmq,

user can auth and have permission to vhost dev

**Comment by michaelklishin** [comment, marked as answer]
Virtual hosts per se do not exist for the plugin as a separate record. It checks whether a user can access a virtual host by executing a [user-provided virtual host access query](https://rabbitmq.com/ldap.html). What that should do is usually dictated by your LDAP directory structure. If you don't have any then I'm not sure why you want to use LDAP to begin with.
